src/helpers/react-native-api.ts

```diff
@@ -0,0 +1,19 @@
+import { createLibraryNotSupportedError } from './errors';
+
+export function importTextFromReactNative() {
+  try {
+    const { Text } = require('react-native');
```

Comment:
@thymikee do you know why are we importing RN's `Text`, etc using dynamic `require` syntax, instead of regular ESM `import` at the top of the file?

Reply:
This was to throw a descriptive error when react-native is not a dependency, and to open the door for using Text outside of `react-native`. Can be moved to import now

Reply:
TBH it's not a bad thing. We have a repo mixing web and native, and we've had people importing RNTL by mistake on web for some dedicated things that doesn't pose a problem (we have runtime check for web vs native). Until 11.2, it was fine, now we have a problem because of the `StyleSheet` import in `'isInaccessible` now breaks just importing it.

But I do understand that it's an edge case that probably doesn't matter for most people.

Reply:
I think that having React Native as a peer dep is enough. We do not do such dynamic checks for other peer deps like React or React Test Renderer.

If we would really want to have a check like this then we could add it insider `index.ts` where we hook into test runner `before` and `after` hooks or `render` as it's always used. Both of these seem to be a better place than `*ByText` queries.